docker/history/history.py

```diff
@@ -1,44 +1,66 @@
+import logging
+import json
+import requests
 from datetime import datetime, timedelta
 from steem import Steem
-from pymongo import MongoClient
+from pymongo import MongoClient, UpdateOne
 from pprint import pprint
-import collections
-import time
-import sys
-import os
+from tenacity import retry, stop_after_attempt, wait_exponential, retry_if_exception_type
 import re
-
+import sys
+import collections  # Importing collections for OrderedDict
+from multiprocessing import Pool
 from apscheduler.schedulers.background import BackgroundScheduler
+import itertools
+import os  # Import os module to read environment variables
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+logger = logging.getLogger(__name__)
 
 log_tag = '[History] '
-env_dist = os.environ
-steemd_url = env_dist.get('STEEMD_URL')
-if steemd_url == None or steemd_url == "":
-    steemd_url = 'https://api.steemit.com'
-mongodb_url = env_dist.get('MONGODB')
-if mongodb_url == None or mongodb_url == "":
-    print(log_tag + 'NEED MONGODB')
+
+# Load configuration from config.json if it exists, otherwise from environment variables
+try:
+    with open('config.json') as config_file:
+        config = json.load(config_file)
+    steemd_urls = config.get('STEEMD_URLS', ['https://api.steemit.com'])
+    mongodb_url = config.get('MONGODB')
+except FileNotFoundError:
```

Comment:
I think this part will not execute in docker style. Because you have copyed a config.json into the docker image.

I suggest we could rename the `config.json` to `config.json.example` .

Reply:
yes , users can decide Not to copy config.json , that way they can pass Environment Variables Directly , ! Thank you for the isnights , i will do the now , its just rename ,and update to Readme too